### PR TITLE
add Account Manager Page

### PR DESCRIPTION
**Pull request by sanglevinh**

### CODE REVIEW

File: frontend/src/pages/AccountManager/index.tsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    dispatch(getListUser(params))
+    //eslint-disable-next-line
+  }, [params])
+
+  const handlePage = (event: ChangeEvent<unknown>, page: number) => {
+    if(!listUser) return
+    setParams(`limit=${listUser.limit}&offset=${page - 1}`)
+  }
+
+  const columns = useMemo(() =>
```
Comment (itutu-tienday) [inline frontend/src/pages/AccountManager/index.tsx:43]: Please make it filterable (partial text match) by UserName, Mail,.

\* Described in Screen Design Document v1.5
- https://drive.google.com/drive/folders/1OZgVlIx-ZxP_BUjB0EAvaExYKc3MhaH-
  - 画面設計_v1.5.pptx

**Review by itutu-tienday — CHANGES_REQUESTED** [review]

**Review by ReiHashimoto — CHANGES_REQUESTED** [review]

File: frontend/src/pages/Login/index.tsx [hunk 2]
```diff
@@ -27,7 +27,8 @@ const Login = () => {
     setIsLoading(true)
     dispatch(login(values))
       .unwrap()
-      .then((_) => {
+      .then(async (_) => {
+        await dispatch(getMe())
```
[the remark this answers is not in the document]
Comment (sanglevinh) [reply frontend/src/pages/Login/index.tsx:31]: @ReiHashimoto We use getMe() every time the user logs in again to check their permissions, and we observe that it is only called once.

File: frontend/src/utils/checkRole.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import { UserDTO } from "api/users/UsersApiDTO"
+
+export const isMe = (user?: UserDTO, idUserWorkSpace?: number) => {
```
[the remark this answers is not in the document]
Comment (sanglevinh) [reply frontend/src/utils/checkRole.ts]: sorry, i converted isMe() to isMine()

File: frontend/src/pages/Login/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (ReiHashimoto) [reply frontend/src/pages/Login/index.tsx:31]: @sanglevinh 
My apologies. My check wasn't enough, works fine.

File: frontend/src/store/slice/User/UserSlice.ts [hunk 4]
```diff
@@ -32,9 +40,35 @@ export const userSlice = createSlice({
...
+        (state) => {
+          state.loading = true
+        },
+      )
+      .addMatcher(
+        isAnyOf(getListUser.rejected),
+        (state) => {
+          state.loading = false
+        },
+      )
```
[the remark this answers is not in the document]
Comment (sanglevinh) [reply frontend/src/store/slice/User/UserSlice.ts]: @ReiHashimoto i fixed it

File: frontend/src/api/users/UsersApiDTO.ts [hunk 5]
```diff
@@ -1,7 +1,12 @@
...
-  id: number
-  uid: string
+  uid?: string
   email: string
+  id?: number
+  name?: string
+  organization_id?: number
+  role_id?: number
+  create_at?: string
+  update_at?: string
```
[the remark this answers is not in the document]
Comment (sanglevinh) [reply frontend/src/api/users/UsersApiDTO.ts]: @ReiHashimoto sorry, i fixed it

File: frontend/src/components/Layout/LeftMenu.tsx [hunk 6]
```diff
@@ -31,7 +35,12 @@ const LeftMenu: FC<{ open: boolean; handleDrawerClose: () => void }> = ({
 
   const onClickWorkspaces = () => {
     handleDrawerClose()
-    navigate('/console/workspaces')
+    navigate('/console/workspaces?limit=50&offset=0')
+  }
+
+  const onClickAccountManager = () => {
+    handleDrawerClose()
+    navigate('/console/account-manager?sort=&sort=&limit=50&offset=0')
```
[the remark this answers is not in the document]
Comment (sanglevinh) [reply frontend/src/components/Layout/LeftMenu.tsx]: @ReiHashimoto i fixed it, I encounter the same workspace with a similar path: '/console/workspaces?limit=50&offset=0'. Would you like me to change it to '/console/workspaces'?

File: frontend/src/pages/AccountManager/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (sanglevinh) [reply frontend/src/pages/AccountManager/index.tsx:109]: @itutu-tienday We have added this section, please check it out for us.

Comment (sanglevinh) [reply frontend/src/pages/AccountManager/index.tsx:109]: @itutu-tienday We have added this section, please check it out for us.

File: frontend/src/components/Layout/LeftMenu.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (ReiHashimoto) [reply frontend/src/components/Layout/LeftMenu.tsx]: @sanglevinh 
Thank you for your implementation.

As you describe, I want the same fix to be applied  to `/console/workspaces`, either.

However, wIth just removing query parameters, api seems to be called twice.
- `/admin/users?limit=50&offset=0`
- `/admin/users?sort=&sort=&limit=50&offset=0`

Let's address these fix in a separate Issue.
So, could you put url back to `'/console/account-manager?sort=&sort=&limit=50&offset=0'` again for now?

[the remark this answers is not in the document]
Comment (sanglevinh) [reply frontend/src/components/Layout/LeftMenu.tsx:43]: @ReiHashimoto, i fixed it

File: frontend/src/@types/index.ts [hunk 7]
```diff
@@ -1,6 +1,6 @@
 export const enum ROLE {
   ADMIN = 1,
-  MANAGER = 10,
+  DATA_MANAGER = 10,
```
Comment (ReiHashimoto) [inline frontend/src/@types/index.ts:3]: Apply change to
- `frontend/src/store/slice/User/UserSelector.ts`

**Review by ReiHashimoto — CHANGES_REQUESTED** [review]

File: frontend/src/pages/AccountManager/index.tsx (hunk 1 above)
Comment (itutu-tienday) [reply frontend/src/pages/AccountManager/index.tsx:109]: @sanglevinh 
Thank you. I will mention a few things I noticed.

- Sorting does not seem to be working in Role column. (API seems to be returning an error 400)
- The filter condition for UserName and Mail should be contain only. 
  - -> Similar to the ExperimentID filter in the Database screen.

File: frontend/src/@types/index.ts (hunk 7 above)
Comment (sanglevinh) [reply frontend/src/@types/index.ts:3]: @ReiHashimoto i fixed it

File: frontend/src/pages/AccountManager/index.tsx (hunk 1 above)
Comment (sanglevinh) [reply frontend/src/pages/AccountManager/index.tsx:111]: @itutu-tienday 

- sorting: I checked and reported BE to correct it
- filter: I fixed it to only be able to filter by contains
Please kindly recheck it.

**Review by ReiHashimoto — APPROVED** [review]
LGTM

**Review by ReiHashimoto — CHANGES_REQUESTED** [review]
sorry, I’ve overlooked.

access to `/console/account-manager` directory, we can see the page.
tables are empty, but non-admin users should be redirected to /console.

**Comment by sanglevinh** [comment]
> sorry, I’ve overlooked.
> 
> access to `/console/account-manager` directory, we can see the page. tables are empty, but non-admin users should be redirected to /console.
@ReiHashimoto 
I fixed it.

**Review by ReiHashimoto — APPROVED** [review]
LGTM

**Review by itutu-tienday — APPROVED** [review]

**Pull request merged**